addons/account/report/account_invoice_report.py

```diff
@@ -98,7 +98,18 @@ def _select(self):
                 move.invoice_payment_term_id,
                 move.invoice_partner_bank_id,
                 move.amount_residual_signed                                 AS residual,
-                move.amount_total_signed                                    AS amount_total,
+                line.price_total / COALESCE(
+                  (SELECT rate FROM res_currency_rate cr WHERE
+                    cr.currency_id = line.currency_id AND
+                    cr.company_id = line.company_id AND
+                    cr.name <= COALESCE(line.date,NOW())
+                  ORDER BY cr.name DESC
+                  LIMIT 1)
+                  ,1) *
+                   (CASE WHEN move.amount_residual_signed < 0
```

Comment:
```suggestion
                   (CASE WHEN move.amount_total_signed < 0
```

As discuss the residual can be 0 (if you pay your bill for example)

Reply:
Maybe you should round after multiplying by the rate.

Reply:
ok, added the changes, the side effect is that now the query is a little mess, but still it seems to work